File: flutter-app/lib/features/mypage/models/settings_response.dart
/// 사용자 설정 응답 모델.
class SettingsData {
  final bool notificationEnabled;
  final String fontSize;
  final bool musicEnabled;
  final int musicVolume;
  final String musicCategory;

  SettingsData({
    required this.notificationEnabled,
    required this.fontSize,
    required this.musicEnabled,
    required this.musicVolume,
    required this.musicCategory,
  });

  factory SettingsData.fromJson(Map<String, dynamic> json) {
    return SettingsData(
      notificationEnabled: json['notificationEnabled'] as bool? ?? true,
      fontSize: json['fontSize'] as String? ?? 'MEDIUM',
      musicEnabled: json['musicEnabled'] as bool? ?? true,
      musicVolume: (json['musicVolume'] as num?)?.toInt() ?? 70,
      musicCategory: json['musicCategory'] as String? ?? 'BGM',
    );
  }
}

File: qtai-server/src/main/java/com/qtai/domain/member/api/dto/SettingsResponse.java
package com.qtai.domain.member.api.dto;

/**
 * 사용자 설정 응답 DTO.
 *
 * @param notificationEnabled 알림 수신 여부
 * @param fontSize 폰트 크기 (SMALL / MEDIUM / LARGE)
 * @param musicEnabled 배경음악 켜기/끄기
 * @param musicVolume 배경음악 볼륨 (0~100)
 * @param musicCategory 배경음악 재생 대상 (ALL / BGM / HYMN)
 */
public record SettingsResponse(
        Boolean notificationEnabled,
        String fontSize,
        Boolean musicEnabled,
        Integer musicVolume,
        String musicCategory
) {}
